client_cli/src/main.rs

```diff
@@ -186,10 +217,10 @@ fn main() -> Result<()> {
 #[allow(clippy::shadow_unrelated)]
 pub fn submit(
```

Comment:
Btw, it seems that `context` is used only within `submit` and `listen` functions. So, maybe you can define a "miniclient" struct that has `submit` and `listen` functions, and does the printing internally?

```rust
trait MiniClient {
  fn listen(&mut self, height: NonZeroU64) -> Result<()>;

  fn submit(&mut self, isi: impl Into<Executable>, metadata: UnlimitedMetadata) -> Result<()>;
}
```

`RunArgs` might work with `MiniClient` trait. The struct that implements `MiniClient` might contain both configuration and debugging logic, which `RunArgs` does not care about.

Reply:
I should notice that `context` is also used in queries + `listen` for blocks/events functions are not the same.

Reply:
Created #3611.